// @zegocloud/zego-uikit-rn/lib/commonjs/signal_plugin/core/in_room_message_core.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _zegoZimReactNative = require("zego-zim-react-native");
var _logger = require("../utils/logger");
var _ZegoUIKitCorePlugin = _interopRequireDefault(require("../../components/internal/ZegoUIKitCorePlugin"));
var _class;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return typeof key === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (typeof input !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (typeof res !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
class ZegoPluginRoomMessageCore {
  constructor() {
    _defineProperty(this, "_onInRoomTextMessageReceivedCallbackMap", {});
    _defineProperty(this, "_onInRoomCommandMessageReceivedCallbackMap", {});
    if (!ZegoPluginRoomMessageCore.shared) {
      ZegoPluginRoomMessageCore.shared = this;
    }
    return ZegoPluginRoomMessageCore.shared;
  }
  static getInstance() {
    if (!ZegoPluginRoomMessageCore.shared) {
      ZegoPluginRoomMessageCore.shared = new ZegoPluginRoomMessageCore();
    }
    return ZegoPluginRoomMessageCore.shared;
  }
  _registerEngineCallback() {
    _ZegoUIKitCorePlugin.default.getZIMPlugin().default.getInstance().on('receiveRoomMessage', (zim, _ref) => {
      let {
        messageList,
        fromConversationID
      } = _ref;
      (0, _logger.zloginfo)('[Core][receiveRoomMessage callback]', messageList, fromConversationID);
      for (const message of messageList) {
        if (message.type === _zegoZimReactNative.ZIMMessageType.Command) {
          const command = message;
          var commandMessage;
          if (command.message instanceof Uint8Array) {
            commandMessage = command.message;
          } else {
            commandMessage = Object.values(command.message);
          }
          var jsonText = decodeURIComponent(escape(String.fromCharCode(...Array.from(commandMessage))));
          const notifyData = {
            message: jsonText,
            roomID: fromConversationID,
            senderUserID: message.senderUserID,
            timestamp: message.timestamp
          };
          this._notifyInRoomCommandMessageReceived(notifyData);
        } else if (message.type === _zegoZimReactNative.ZIMMessageType.Text) {
          const textMessage = message;
          const notifyData = {
            message: textMessage.message,
            roomID: fromConversationID,
            senderUserID: message.senderUserID,
            timestamp: message.timestamp
          };
          this._notifyInRoomTextMessageReceived(notifyData);
        }
      }
    });
    (0, _logger.zloginfo)('[ZegoPluginRoomMessageCore]Register callback for ZIM...');
  }
  _unregisterEngineCallback() {
    (0, _logger.zloginfo)('[ZegoPluginRoomMessageCore]Unregister callback from ZIM...');
    _ZegoUIKitCorePlugin.default.getZIMPlugin().default.getInstance().off('receiveRoomMessage');
  }
  _notifyInRoomTextMessageReceived(notifyData) {
    (0, _logger.zloginfo)(`[Core]NotifyInRoomTextMessageReceived, data: ${notifyData}`);
    Object.keys(this._onInRoomTextMessageReceivedCallbackMap).forEach(callbackID => {
      if (this._onInRoomTextMessageReceivedCallbackMap[callbackID]) {
        this._onInRoomTextMessageReceivedCallbackMap[callbackID](notifyData);
      }
    });
  }
  _notifyInRoomCommandMessageReceived(notifyData) {
    (0, _logger.zloginfo)(`[Core]NotifyInRoomCommandMessageReceived, data: ${notifyData}`);
    Object.keys(this._onInRoomCommandMessageReceivedCallbackMap).forEach(callbackID => {
      if (this._onInRoomCommandMessageReceivedCallbackMap[callbackID]) {
        this._onInRoomCommandMessageReceivedCallbackMap[callbackID](notifyData);
      }
    });
  }
  sendInRoomTextMessage(roomID, message) {
    return new Promise((resolve, reject) => {
      const textMessage = {
        type: _zegoZimReactNative.ZIMMessageType.Text,
        message: message
      };
      _ZegoUIKitCorePlugin.default.getZIMPlugin().default.getInstance().sendMessage(textMessage, roomID, _zegoZimReactNative.ZIMConversationType.Room, {
        priority: 1
      }).then(_ref2 => {
        let {
          message
        } = _ref2;
        (0, _logger.zloginfo)(`[Core]sendInRoomTextMessage done, roomID: ${roomID}, message: ${message}`);
        resolve();
      }).catch(error => {
        reject(error);
      });
    });
  }
  sendInRoomCommandMessage(roomID, message) {
    var command = new Uint8Array(Array.from(unescape(encodeURIComponent(message))).map(val => val.charCodeAt(0)));
    return new Promise((resolve, reject) => {
      const commandMessage = {
        type: _zegoZimReactNative.ZIMMessageType.Command,
        message: command
      };
      _ZegoUIKitCorePlugin.default.getZIMPlugin().default.getInstance().sendMessage(commandMessage, roomID, _zegoZimReactNative.ZIMConversationType.Room, {
        priority: 1
      }).then(_ref3 => {
        let {
          message
        } = _ref3;
        (0, _logger.zloginfo)(`[Core]sendInRoomCommandMessage done, roomID: ${roomID}, message: ${message}`);
        resolve();
      }).catch(error => {
        reject(error);
      });
    });
  }
  onInRoomTextMessageReceived(callbackID, callback) {
    if (!_ZegoUIKitCorePlugin.default.getZIMPlugin().default.getInstance()) {
      (0, _logger.zlogerror)('[ZegoPluginRoomPropertiesCore]Please initialize it first.');
    }
    if (typeof callback !== 'function') {
      if (callbackID in this._onInRoomTextMessageReceivedCallbackMap) {
        (0, _logger.zloginfo)('[Core][onRoomPropertyUpdated] Remove callback for: [', callbackID, '] because callback is not a valid function!');
        delete this._onInRoomTextMessageReceivedCallbackMap[callbackID];
      }
    } else {
      this._onInRoomTextMessageReceivedCallbackMap[callbackID] = callback;
    }
  }
  onInRoomCommandMessageReceived(callbackID, callback) {
    if (typeof callback !== 'function') {
      if (callbackID in this._onInRoomCommandMessageReceivedCallbackMap) {
        (0, _logger.zloginfo)('[Core][onInRoomCommandMessageReceived] Remove callback for: [', callbackID, '] because callback is not a valid function!');
        delete this._onInRoomCommandMessageReceivedCallbackMap[callbackID];
      }
    } else {
      this._onInRoomCommandMessageReceivedCallbackMap[callbackID] = callback;
    }
  }
}
exports.default = ZegoPluginRoomMessageCore;
_class = ZegoPluginRoomMessageCore;
_defineProperty(ZegoPluginRoomMessageCore, "shared", void 0);
