// @zegocloud/zego-uikit-rn/lib/commonjs/components/audio_video/ZegoToggleCameraButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ZegoToggleCameraButton;
var _react = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _ZegoUIKitInternal = _interopRequireDefault(require("../internal/ZegoUIKitInternal"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function ZegoToggleCameraButton(props) {
  const {
    userID,
    iconCameraOn,
    iconCameraOff,
    isOn,
    onPress,
    width = 48,
    height = 48
  } = props;
  const [isCurrentOn, setIsCurrentOn] = (0, _react.useState)(true); // Default on
  const getImageSourceByPath = () => {
    const pathOn = iconCameraOn ? iconCameraOn : require("../internal/resources/white_button_camera_on.png");
    const pathOff = iconCameraOff ? iconCameraOff : require("../internal/resources/white_button_camera_off.png");
    return isCurrentOn ? pathOn : pathOff;
  };
  const onButtonPress = () => {
    _ZegoUIKitInternal.default.turnCameraDeviceOn(userID, !isCurrentOn);
    if (typeof onPress == 'function') {
      onPress();
    }
  };
  (0, _react.useEffect)(() => {
    setIsCurrentOn(_ZegoUIKitInternal.default.isCameraDeviceOn(userID));
  });
  (0, _react.useEffect)(() => {
    const callbackID = 'ZegoToggleCameraButton' + String(Math.floor(Math.random() * 10000));
    _ZegoUIKitInternal.default.onCameraDeviceOn(callbackID, (id, on) => {
      if (userID === undefined || userID === '') {
        // local user
        if (id == _ZegoUIKitInternal.default.getLocalUserInfo().userID) {
          setIsCurrentOn(on);
        }
      } else if (id == userID) {
        setIsCurrentOn(on);
      }
    });
    _ZegoUIKitInternal.default.onSDKConnected(callbackID, () => {
      _ZegoUIKitInternal.default.turnCameraDeviceOn(userID, isOn);
      setIsCurrentOn(_ZegoUIKitInternal.default.isCameraDeviceOn(userID));
    });
    return () => {
      _ZegoUIKitInternal.default.onCameraDeviceOn(callbackID);
      _ZegoUIKitInternal.default.onSDKConnected(callbackID);
    };
  }, []);
  return /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    style: {
      width: width,
      height: height
    },
    onPress: onButtonPress
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Image, {
    resizeMode: "contain",
    source: getImageSourceByPath(),
    style: {
      width: "100%",
      height: "100%"
    }
  }));
}
