// @zegocloud/zego-uikit-rn/lib/commonjs/components/audio_video/ZegoSwitchCameraButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ZegoSwitchCameraButton;
var _react = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _ZegoUIKitInternal = _interopRequireDefault(require("../internal/ZegoUIKitInternal"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function ZegoSwitchCameraButton(props) {
  // TODO useFrontFacingCamera may cause problems when create a lot of times during connected
  const {
    iconFrontFacingCamera,
    iconBackFacingCamera,
    useFrontFacingCamera = true,
    onPress,
    width = 48,
    height = 48
  } = props;
  const [isFront, setIsFront] = (0, _react.useState)(_ZegoUIKitInternal.default.isUsingFrontFacingCamera());
  const getImageSourceByPath = () => {
    const pathFront = iconFrontFacingCamera ? iconFrontFacingCamera : require("../internal/resources/white_button_flip_camera.png");
    const pathBack = iconBackFacingCamera ? iconFrontFacingCamera : require("../internal/resources/white_button_flip_camera.png");
    return isFront ? pathFront : pathBack;
  };
  const onButtonPress = () => {
    _ZegoUIKitInternal.default.useFrontFacingCamera(!isFront);
    setIsFront(!isFront);
    if (typeof onPress == 'function') {
      onPress();
    }
  };
  (0, _react.useEffect)(() => {
    const callbackID = 'ZegoSwitchCameraButton' + String(Math.floor(Math.random() * 10000));
    _ZegoUIKitInternal.default.onSDKConnected(callbackID, () => {
      _ZegoUIKitInternal.default.useFrontFacingCamera(useFrontFacingCamera);
      setIsFront(useFrontFacingCamera);
    });
    return () => {
      _ZegoUIKitInternal.default.onSDKConnected(callbackID);
    };
  }, []);
  return /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    style: {
      width: width,
      height: height
    },
    onPress: onButtonPress
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Image, {
    resizeMode: "contain",
    source: getImageSourceByPath(),
    style: {
      width: "100%",
      height: "100%"
    }
  }));
}
