// @zegocloud/zego-uikit-rn/lib/module/signal_plugin/services/in_room_message_service.js
var _class;
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return typeof key === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (typeof input !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (typeof res !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
import ZegoPluginRoomMessageCore from '../core/in_room_message_core';
export default class ZegoPluginInRoomMessageService {
  constructor() {
    if (!ZegoPluginInRoomMessageService.shared) {
      ZegoPluginInRoomMessageService.shared = this;
    }
    return ZegoPluginInRoomMessageService.shared;
  }
  static getInstance() {
    if (!ZegoPluginInRoomMessageService.shared) {
      ZegoPluginInRoomMessageService.shared = new ZegoPluginInRoomMessageService();
    }
    return ZegoPluginInRoomMessageService.shared;
  }
  sendInRoomTextMessage(roomID, message) {
    return ZegoPluginRoomMessageCore.getInstance().sendInRoomTextMessage(roomID, message);
  }
  sendInRoomCommandMessage(roomID, message) {
    return ZegoPluginRoomMessageCore.getInstance().sendInRoomCommandMessage(roomID, message);
  }
  onInRoomTextMessageReceived(callbackID, callback) {
    ZegoPluginRoomMessageCore.getInstance().onInRoomTextMessageReceived(callbackID, callback);
  }
  onInRoomCommandMessageReceived(callbackID, callback) {
    ZegoPluginRoomMessageCore.getInstance().onInRoomCommandMessageReceived(callbackID, callback);
  }
}
_class = ZegoPluginInRoomMessageService;
_defineProperty(ZegoPluginInRoomMessageService, "shared", void 0);
