// @zegocloud/zego-uikit-rn/lib/commonjs/signal_plugin/services/in_room_message_service.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _in_room_message_core = _interopRequireDefault(require("../core/in_room_message_core"));
var _class;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return typeof key === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (typeof input !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (typeof res !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
class ZegoPluginInRoomMessageService {
  constructor() {
    if (!ZegoPluginInRoomMessageService.shared) {
      ZegoPluginInRoomMessageService.shared = this;
    }
    return ZegoPluginInRoomMessageService.shared;
  }
  static getInstance() {
    if (!ZegoPluginInRoomMessageService.shared) {
      ZegoPluginInRoomMessageService.shared = new ZegoPluginInRoomMessageService();
    }
    return ZegoPluginInRoomMessageService.shared;
  }
  sendInRoomTextMessage(roomID, message) {
    return _in_room_message_core.default.getInstance().sendInRoomTextMessage(roomID, message);
  }
  sendInRoomCommandMessage(roomID, message) {
    return _in_room_message_core.default.getInstance().sendInRoomCommandMessage(roomID, message);
  }
  onInRoomTextMessageReceived(callbackID, callback) {
    _in_room_message_core.default.getInstance().onInRoomTextMessageReceived(callbackID, callback);
  }
  onInRoomCommandMessageReceived(callbackID, callback) {
    _in_room_message_core.default.getInstance().onInRoomCommandMessageReceived(callbackID, callback);
  }
}
exports.default = ZegoPluginInRoomMessageService;
_class = ZegoPluginInRoomMessageService;
_defineProperty(ZegoPluginInRoomMessageService, "shared", void 0);
