// @zegocloud/zego-uikit-rn/lib/commonjs/components/audio_video/ZegoAudioVideoView/VideoFrame.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = VideoFrame;
var _react = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _zegoExpressEngineReactnative = require("zego-express-engine-reactnative");
var _ZegoUIKitInternal = _interopRequireDefault(require("../../internal/ZegoUIKitInternal"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function VideoFrame(props) {
  const {
    userID,
    roomID,
    fillMode
  } = props;
  const viewRef = (0, _react.useRef)(null);
  console.log('VideoFrame', userID);
  const updateRenderingProperty = () => {
    console.log('VideoFrame updateRenderingProperty');
    const viewID = (0, _reactNative.findNodeHandle)(viewRef.current);
    _ZegoUIKitInternal.default.updateRenderingProperty(userID, viewID, fillMode);
  };
  (0, _react.useEffect)(() => {
    updateRenderingProperty();
    const callbackID = 'VideoFrame' + userID + String(Math.floor(Math.random() * 10000));
    _ZegoUIKitInternal.default.onSDKConnected(callbackID, () => {
      updateRenderingProperty();
    });
    _ZegoUIKitInternal.default.onUserJoin(callbackID, userInfoList => {
      userInfoList.forEach(user => {
        if (user.userID == userID) {
          updateRenderingProperty();
        }
      });
    });
    _ZegoUIKitInternal.default.onVideoViewForceRender(callbackID, () => {
      updateRenderingProperty();
    });
    return () => {
      _ZegoUIKitInternal.default.onSDKConnected(callbackID);
      _ZegoUIKitInternal.default.onUserJoin(callbackID);
      _ZegoUIKitInternal.default.onVideoViewForceRender(callbackID);
      _ZegoUIKitInternal.default.updateRenderingProperty(userID, -1, fillMode);
    };
  }, []);
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: styles.container
  }, /*#__PURE__*/_react.default.createElement(_zegoExpressEngineReactnative.ZegoTextureView
  // @ts-ignore
  , {
    style: styles.videoContainer,
    ref: viewRef,
    collapsable: false
  }), /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: styles.audioContainer
  }, props.children));
}
const styles = _reactNative.StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    height: '100%',
    position: 'absolute'
  },
  videoContainer: {
    // flex: 1,
    width: '100%',
    height: '100%',
    top: 0,
    left: 0,
    position: 'absolute',
    zIndex: 1
  },
  audioContainer: {
    width: '100%',
    height: '100%',
    top: 0,
    left: 0,
    position: 'absolute',
    zIndex: 2
  }
});
